test/e2e/llmisvc/test_llm_inference_service.py

```diff
@@ -54,12 +98,21 @@ def assert_200_with_choices(response: requests.Response) -> None:
     ), f"Expected 200 with choices, got {response.status_code}: {response.text}"
 
 
+@dataclass
+class RouterResources:
```

Comment:
Maybe this can be a part of `TestCase` dataclass instead? `BeforeTest`  field that keeps list of functions that can create needed resources. Similar to assertions.

We can establish ownership through owner reference so when LLMInferenceService is deleted, all the resources are going to be garbage collected?